compiler/src/iree/compiler/Codegen/Common/LinkTuningSpecsPass.cpp

```diff
@@ -138,6 +383,85 @@ emitLinkedTuningSpec(ModuleOp module, ArrayRef<NamedSequenceOp> specsToLink) {
   return newSpec;
 }
 
+static FailureOr<NamedSequenceOp> emitLinkedDefaultTuningSpec(ModuleOp module) {
+  OpBuilder builder(module.getContext());
+
+  // Step 1: Collect NamedSequenceOps from inner modules.
+  TuningSpecsToMerge tuningSpecs = collectTuningSpecsToMerge(module);
+  SmallVector<NamedSequenceOp> &namedSequenceOpsToMove =
+      tuningSpecs.namedSequenceOpsToMove;
+  llvm::DenseMap<NamedSequenceOp, ForeachMatchOp> &namedSequenceToUser =
+      tuningSpecs.namedSequenceToUser;
+
+  // Step 2-a: Make sure the name sequence names are unique, and then move
+  // collected NamedSequenceOps to the top-level module.
+  resolveAndMoveNamedSequenceOps(namedSequenceOpsToMove, module, builder,
+                                 namedSequenceToUser);
+
+  // Step 2-b: Create a new NamedSequenceOp `__kernel_config` in the top-level
+  // module.
+
+  // Collect all `ForeachMatchOp`s from `__kernel_config` NamedSequenceOps
+  // in inner modules to merge into the new `__kernel_config` in the top module.
```

Comment:
I think something like this would be more descriptive:
```suggestion
  // Collect the `ForeachMatchOp`s from the entry point named sequences of
  // each inner module to merge into the new default entry point in the top
  // module.
```